src/finetune.py

```diff
@@ -122,6 +129,11 @@ def finetune_groupwise(
 
         if args.finetune_relative_mse_tolerance is not None:
             epoch_loss = loss_numerator / loss_denominator
+            if args.finetune_keep_best:
+                if epoch_loss / previous_best_loss < 1.0:
+                    best_parameters = deepcopy(differentiable_parameters)
+                else:
+                    differentiable_parameters = best_parameters
```

Comment:
What's the point of this? If loss increases we roll back to best?

Reply:
Yes, seems to bring marginal improvement.